### PR TITLE
svd2rust: 0.14.0 -> 0.17.0

### PR DESCRIPTION
**Pull request by arcnmx**
###### Motivation for this change

Bring this up-to-date with the latest rust-embedded crates. 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 121102` at 742dcc62 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>svd2rust</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/maintainers-missing.md">maintainers-missing</a>

Package does not have a maintainer. Consider adding yourself?

Near pkgs/build-support/rust/default.nix:144:3:
```
    |
144 |   meta = {
    |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/tools/rust/svd2rust/default.nix [hunk 1]
```diff
@@ -4,17 +4,17 @@ with rustPlatform;
...
-    sha256 = "1a0ldmjkhyv5c52gcq8p8avkj0cgj1b367w6hm85bxdf5j4y8rra";
+    sha256 = "0p118dbxnypp99xn2xa382rvwk56a9p4wc6xhpfn8qbg4g43brrd";
   };
   cargoPatches = [ ./cargo-lock.patch ];
 
-  cargoSha256 = "0n0xc8b982ra007l6gygssf1n60gfc2rphwyi7n95dbys1chciyg";
+  cargoSha256 = "15js131shsqc2f42ixpp9rhmqp08cy03m4fx5l0vp1y6ikhqrqz4";
 
   # doc tests fail due to missing dependency
   doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/tools/rust/svd2rust/default.nix:20]: Can we try to enable tests?

Comment (arcnmx) [reply pkgs/development/tools/rust/svd2rust/default.nix]: Re-enabled tests, seems to work fine. Not sure about maintainership though, I don't use it *that* frequently.

**Comment by SuperSandro2000** [comment]
We just landed 0.18 in master.

**Comment by arcnmx** [comment]
I'm feeling like this package should actually be removed from nixpkgs, because 0.18 is useless until other codebases update to fix the breakage caused by that update.

The problem is that any codebase using svd2rust depends on a *very specific version* of it, and right now that's not even the latest 0.18, so... unless nixpkgs wants to carry all versions going forward (like svd2rust_17, etc), it's kind of pointless? tbh I'm not sure why it's even in nixpkgs, because it's pretty inappropriate to have without pinned versions of it.

**Comment by SuperSandro2000** [comment]
> unless nixpkgs wants to carry all versions going forward (like svd2rust_17, etc), it's kind of pointless?

I don't know. I never used the tool.

> Does anyone agree that adding pinned version attrs would be appropriate?

I ping @astro about it

**Comment by astro** [comment]
I'd argue that this tool isn't much used as a dependency of other packages. Instead, you use it once to generate a new PAC crate for an MCU. People upload these to crates.io then where they can be used without running svd2rust again.

**Pull request closed without merging**